### ISSUE TITLE
[Support] Need explanation for "Conclusion: don't install ..."

### ISSUE THREAD

**Issue by antonioribeiro** [composer/composer#2702]
Sorry to open an issue for this, but I can't understand why Composer isn't installing a dependency:

```
$ composer update --prefer-source
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - pragmarx/firewall dev-master requires pragmarx/support dev-master -> satisfiable by pragmarx/support[dev-master].
    - pragmarx/firewall dev-master requires pragmarx/support dev-master -> satisfiable by pragmarx/support[dev-master].
    - Conclusion: don't install pragmarx/support dev-master
    - Installation request for pragmarx/firewall dev-master -> satisfiable by pragmarx/firewall[dev-master].
```

This is my composer.json:

```
{
    "name": "laravel/laravel",
    "description": "The Laravel Framework.",
    "keywords": ["framework", "laravel"],
    "license": "MIT",
    "require": {
        "laravel/framework": "4.2.*",
        "cartalyst/sentry": "2.1.*",
        "way/generators": "dev-master",
        "raveren/kint": "dev-master",
        "zackkitzmiller/tiny": "1.0.1",
        "pragmarx/glottos": "dev-master",
        "pragmarx/firewall": "dev-master",
        "davejamesmiller/laravel-breadcrumbs": "dev-master",
        "way/form": "dev-master"
    },
    "autoload": {
        "classmap": [
            "app/commands",
            "app/database/migrations",
            "app/database/seeds",
            "app/tests/TestCase.php"
        ],
        "psr-0": {
            "App": "app/"
        }
    },
    "scripts": {
        "post-install-cmd": [
            "php artisan optimize"
        ],
        "post-update-cmd": [
            "php artisan clear-compiled",
            "php artisan optimize"
        ],
        "post-create-project-cmd": [
            "php artisan key:generate"
        ]
    },
    "config": {
        "preferred-install": "dist"
    },
    "minimum-stability": "dev"
}
```

**Comment by XedinUnknown** [comment]
This doesn't really explain why this is happening. I have the same problem, but the deps are my own. Can't just switch to version: there isn't a release.

**Comment by stof** [comment]
I suggest you to read about the minimum-stability in the documentation.

**Comment by XedinUnknown** [comment]
I have it set to `dev`.

**Comment by XedinUnknown** [comment]
I resolved the problem. One of the packages, that was not listed in the error message, depended on another package, that was not listed in the error message. That second package was of a version different to what the one in the error message requires. But there was no way to know about it from the message. So, what I did was I removed all dependencies, then added them back one by one, running `composer update` after each change, and fixing the problems in dep packages along the way. Hope this method helps somebody.
